source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -582,17 +566,16 @@ ur_result_t ur_queue_immediate_in_order_t::enqueueMemBufferMap(
 
   auto waitList = getWaitListView(phEventWaitList, numEventsInWaitList);
 
-  bool memoryMigrated = false;
   auto pDst = ur_cast<char *>(hBuffer->mapHostPtr(
       mapFlags, offset, size, [&](void *src, void *dst, size_t size) {
         ZE2UR_CALL_THROWS(zeCommandListAppendMemoryCopy,
                           (commandListManager.getZeCommandList(), dst, src,
                            size, nullptr, waitList.second, waitList.first));
-        memoryMigrated = true;
+        clearWaitList(waitList);
       }));
   *ppRetMap = pDst;
 
-  if (!memoryMigrated && waitList.second) {
+  if (!waitList.second) {
```

Comment:
I had to scroll up to see what `second` is to understand that this isn't a null-check. Either use `!= 0` to make that explicit or use a struct with variable names.

Reply:
It doesn't really matter since you can check any of the member, but I changed it to be less confusing.

Reply:
it matters  because if you were checking the pointer, i'd be `if (waitList.first)` without negation. I think that's an easy mistake to make (and what I initially thought was happening here).
Anyway, looks good now, thanks.

Reply:
actually, I think I just made a mistake in the previous revision - there shouldn't be any negation here